influence_benchmark/environment/environment.py

```diff
@@ -9,6 +9,69 @@
 from influence_benchmark.root import PROJECT_ROOT
 
 
+class VecEnv:
+    def __init__(self, envs):
+        self.envs = envs
+        # We only really need all backends to be the same. I think we should move the backend
+        # to be a environment attribute that then is passed to the preference model and so on.
+        assert all(self.envs[0].config == env.config for env in self.envs)
+        assert all(self.envs[0].extra_configs == env.extra_configs for env in self.envs)
+
+    def reset(self):
+        return [env.reset() for env in self.envs]
+
+    def step_vec(self, action_n):
+        state_n = [env.current_state for env in self.envs]
+
+        # Vectorized transition model
+        next_state_n = self.vectorized_transition(action_n, state_n)
+
+        self.vectorized_preference_transition(action_n, state_n, next_state_n)
+
+        for env, state, transition, next_state in zip(self.envs, state_n, action_n, next_state_n):
+            env.get_env_char_response(state, transition, next_state)
+            env.current_state = next_state
+            done = env.is_terminal(env.current_state)
+
+        return next_state_n, [done for _ in self.envs]
+
+    def vectorized_preference_transition(self, action_n, state_n, next_state_n):
```

Comment:
It's probably not wrong but I don't understand why you need the next state here?

Reply:
It appears on line 47 i think

Reply:
But we didn't need it for the single environment?

Reply:
Yeah you're right, there's something fishy going on there

Reply:
Oh, it's just because in this function we're also doing the variable assignment:

`next_state.preferences = self.preference_model.get_preferences(state, action)`

which requires `next_state` to be defined within the scope